src/plugins/backend_odbc/backend_odbc_get.c

```diff
@@ -0,0 +1,842 @@
+					retGetLongData =
+						getLongData (sqlStmt, 2, SQL_C_CHAR, &longKeyString, KEYSTRING_BUFFER_SIZE * 2, errorKey);
+				}
+				else if (!isFurtherMetaKey && KEYSTRING_BUFFER_SIZE <= buffers->strLenInd)
+				{
+					longKeyString = (char *) elektraMalloc (sizeof (char) * (buffers->strLenInd + 1));
+					retGetLongData =
+						getLongData (sqlStmt, 2, SQL_C_CHAR, &longKeyString, (buffers->strLenInd + 1), errorKey);
+				}
+
+				/* Check metakey-name column */
+				if (retGetLongData && buffers->metaNameLenInd == SQL_NO_TOTAL)
+				{
+					longMetaKeyName = (char *) elektraMalloc (sizeof (char) * (METAKEYNAME_BUFFER_SIZE * 2));
+					retGetLongData = getLongData (sqlStmt, 3, SQL_C_CHAR, &longMetaKeyName, METAKEYNAME_BUFFER_SIZE * 2,
+								      errorKey);
+				}
+				else if (METAKEYNAME_BUFFER_SIZE <= buffers->metaNameLenInd)
+				{
+					longMetaKeyName = (char *) elektraMalloc (sizeof (char) * (buffers->metaNameLenInd + 1));
+					retGetLongData = getLongData (sqlStmt, 3, SQL_C_CHAR, &longMetaKeyName,
+								      (buffers->metaNameLenInd + 1), errorKey);
+				}
+
+				/* Check metakey-string column */
+				if (retGetLongData && buffers->metaStrLenInd == SQL_NO_TOTAL)
+				{
+					longMetaKeyString = (char *) elektraMalloc (sizeof (char) * (METASTRING_BUFFER_SIZE * 2));
+					retGetLongData = getLongData (sqlStmt, 4, SQL_C_CHAR, &longMetaKeyString,
+								      METASTRING_BUFFER_SIZE * 2, errorKey);
+				}
+				else if (METASTRING_BUFFER_SIZE <= buffers->metaStrLenInd)
+				{
+					longMetaKeyString = (char *) elektraMalloc (sizeof (char) * (buffers->metaStrLenInd + 1));
+					retGetLongData = getLongData (sqlStmt, 4, SQL_C_CHAR, &longMetaKeyString,
+								      (buffers->metaStrLenInd + 1), errorKey);
+				}
+
+				if (!retGetLongData)
+				{
+					/* The variable longKeyName should not be freed here, it either points to the same memory as
+					 * prevKeyName or is NULL */
+
+					elektraFree (prevKeyName);
+					elektraFree (longKeyString);
+					elektraFree (longMetaKeyName);
+					elektraFree (longMetaKeyString);
+
+					/* Statement handle was already freed by getLongData() function */
+					if (ksResult)
+					{
+						ksDel (ksResult);
+					}
+
+					SQLFreeHandle (SQL_HANDLE_STMT, sqlStmt);
+					return NULL;
+				}
+			}
+			else
+			{
+				ELEKTRA_ADD_ODBC_WARNING (SQL_HANDLE_STMT, sqlStmt, errorKey);
+			}
+		}
+
+		if (!isFurtherMetaKey && !longKeyName)
+		{
+			/* Check short keyname */
+			if (prevKeyName && elektraStrCmp (prevKeyName, (const char *) buffers->bufferKeyName) == 0)
+			{
+				isFurtherMetaKey = true;
+			}
+			else
+			{
+				elektraFree (prevKeyName);
+				prevKeyName = elektraStrDup ((const char *) buffers->bufferKeyName);
+			}
+		}
+
+		if (!ksResult)
+		{
+			ksResult = ksNew (0, KS_END);
+		}
+
+		if (!isFurtherMetaKey)
+		{
+			/* Create new key */
+			prevKey = curKey;
+			curKey = keyDup (errorKey, KEY_CP_NAME);
```

Comment:
If the keys is used for things other than errors, it shouldn't be called `errorKey`